service_account_getter.go

```diff
@@ -0,0 +1,108 @@
+package kubeauth
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+	kubeerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+const annotationKeyPrefix = "auth-metadata.vault.hashicorp.com/"
+
+// serviceAccountGetter defines a namespace validator interface
+type serviceAccountGetter interface {
+	annotations(context.Context, *http.Client, string, string) (map[string]string, error)
+}
+
+type serviceAccountGetterFactory func(*kubeConfig) serviceAccountGetter
+
+// serviceAccountGetterWrapper implements the serviceAccountGetter interface
+type serviceAccountGetterWrapper struct {
+	config *kubeConfig
+}
+
+func newServiceAccountGetterWrapper(config *kubeConfig) serviceAccountGetter {
+	return &serviceAccountGetterWrapper{
+		config: config,
+	}
+}
+
+func (w *serviceAccountGetterWrapper) annotations(ctx context.Context, client *http.Client, namespace, serviceAccount string) (map[string]string, error) {
+	url := fmt.Sprintf("%s/api/v1/namespaces/%s/serviceaccounts/%s",
+		strings.TrimSuffix(w.config.Host, "/"), namespace, serviceAccount)
+	req, err := http.NewRequestWithContext(ctx, "GET", url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	// Use the configured TokenReviewer JWT as the bearer
+	if w.config.TokenReviewerJWT == "" {
+		return nil, errors.New("service account lookup failed: TokenReviewer JWT needs to be configured to retrieve service accounts")
```

Comment:
Is there any reason we can't use the client's own token like we do for other API requests? It's not my favourite feature, but I would favour consistency and shared code to achieve that if possible.